src/component/modal/editRange/forms/components/SignalsForm.jsx

```diff
@@ -0,0 +1,221 @@
+        setFieldValue(activeField, event.xPPM);
+      }
+    });
+
+    return () => {
+      unsubscribe();
+    };
+  }, [values.activeTab, activeField, setFieldValue]);
+
+  const handleOnFocus = useCallback(
+    (name) => {
+      setActiveField(name);
+    },
+    [setActiveField],
+  );
+
+  const handleOnBlur = useCallback(() => {
+    setActiveField(null);
+  }, []);
+
+  const tapClickHandler = useCallback(
+    ({ tabid }) => {
+      setFieldValue('activeTab', tabid);
+    },
+    [setFieldValue],
+  );
+
+  const onDeleteSignal = useCallback(
+    ({ tabid }) => {
+      const _signals = values.signals.filter(
+        (_signal, i) => i !== Number(tabid),
+      );
+      setFieldValue('signals', _signals);
+      setFieldValue(
+        'activeTab',
+        _signals.length > 0 ? (_signals.length - 1).toString() : 'addSignalTab',
+      );
+    },
+    [setFieldValue, values.signals],
+  );
+
+  const tabContainsErrors = useCallback(
+    (i) => {
+      return (
+        errors.signals &&
+        ((errors.signals.noCouplings &&
+          errors.signals.noCouplings.some((_error) => _error.index === i)) ||
+          (errors.signals[`${i}`] &&
+            errors.signals[`${i}`].missingCouplings &&
+            errors.signals[`${i}`].missingCouplings.length > 0))
+      );
+    },
+    [errors.signals],
+  );
+
+  const signalFormTabs = useMemo(() => {
+    const signalTabs =
+      values.signals.length > 0
+        ? values.signals.map((signal, i) => (
+            <div
+              // eslint-disable-next-line react/no-array-index-key
+              key={`signalForm${i}`}
+              tabid={`${i}`}
+              tablabel={`${'\u0394'}: ${signal.delta.toFixed(
+                5,
+              )} (${signal.j
+                .map((_coupling) => translateMultiplet(_coupling.multiplicity))
+                .join('')})`}
+              tabstyles={tabContainsErrors(i) ? tabStylesAddition : null}
+            >
+              <SignalFormTab onFocus={handleOnFocus} onBlur={handleOnBlur} />
+            </div>
+          ))
+        : [];
+
+    const addSignalTab = (
+      <div
+        tablabel={'\u002B'}
+        tabid="addSignalTab"
+        candelete="false"
+        key="addSignalTab"
+      >
+        <AddSignalFormTab onFocus={handleOnFocus} onBlur={handleOnBlur} />
+      </div>
+    );
+
+    return signalTabs.concat(addSignalTab);
+  }, [handleOnBlur, handleOnFocus, tabContainsErrors, values.signals]);
+
+  const [tabs, setTabs] = useState(
+    <Tabs
+      defaultTabID={values.activeTab}
+      onClick={tapClickHandler}
+      canDelete={true}
+      onDelete={onDeleteSignal}
+    >
+      {signalFormTabs}
+    </Tabs>,
+  );
+
+  useEffect(() => {
+    setTabs(
+      <Tabs
+        defaultTabID={values.activeTab}
+        onClick={tapClickHandler}
+        canDelete={true}
+        onDelete={onDeleteSignal}
+      >
+        {signalFormTabs}
+      </Tabs>,
+    );
```

Comment:
Why you set component inside the state, you have to store only  data and not components inside state.